src/Compilers/CSharp/Portable/Symbols/Source/LambdaSymbol.cs

```diff
@@ -361,7 +361,8 @@ public override bool HidesBaseMethodsByName
                 var name = unboundLambda.ParameterName(p);
                 var location = unboundLambda.ParameterLocation(p);
                 var locations = location == null ? ImmutableArray<Location>.Empty : ImmutableArray.Create<Location>(location);
-                var parameter = new SourceSimpleParameterSymbol(this, type, p, refKind, name, locations);
+
+                var parameter = new SourceSimpleParameterSymbol(owner: this, type, ordinal: p, refKind, name, unboundLambda.ParameterIsDiscard(p), locations);
```

Comment:
>name [](start = 104, length = 4)

It looks like this could be null now. I don't think would want the symbol to return null as its ```Name```. We say: "Returns the empty string if unnamed." Consider adjusting implementation of ```ParameterName``` parameter name to never return null (preferred), or adding a compensation here. Please add a targeted test for behavior of ISymbol.Name/Symbol.Name API.